code/modules/projectiles/guns/energy.dm

```diff
@@ -411,3 +411,125 @@
 
 /obj/item/weapon/gun/energy/lasgun/M43/practice/unique_action(mob/user)
 	return
+
+/obj/item/weapon/gun/energy/lasgun/lasrifle
+	name = "\improper TX-73 Lasrifle MK2"
+	desc = "A multifunctional laser based rifle with an integrated mode selector. Ideal for any situation. Uses power cells instead of ballistic magazines.."
+	force = 20 //Large and hefty! Includes stock bonus.
+	icon = 'icons/Marine/lasrifle.dmi'
+	icon_state = "tx73"
+	item_state = "tx73"
+	max_shots = 50 //codex stuff
+	load_method = CELL //codex stuff
+	ammo = /datum/ammo/energy/lasgun/M43
+	ammo_diff = null
+	cell_type = null
+	charge_cost = 20
+	attachable_allowed = list(
+						/obj/item/attachable/bayonet,
+						/obj/item/attachable/reddot,
+						/obj/item/attachable/verticalgrip,
+						/obj/item/attachable/angledgrip,
+						/obj/item/attachable/lasersight,
+						/obj/item/attachable/gyro,
+						/obj/item/attachable/flashlight,
+						/obj/item/attachable/bipod,
+						/obj/item/attachable/magnetic_harness,
+						/obj/item/attachable/attached_gun/grenade,
+						/obj/item/attachable/scope,
+						/obj/item/attachable/attached_gun/flamer,
+						/obj/item/attachable/attached_gun/shotgun,
+						/obj/item/attachable/scope/mini)
+	flags_gun_features = GUN_AUTO_EJECTOR|GUN_CAN_POINTBLANK|GUN_AMMO_COUNTER|GUN_ENERGY|GUN_AMMO_COUNTER
+	attachable_offset = list("muzzle_x" = 32, "muzzle_y" = 18,"rail_x" = 12, "rail_y" = 23, "under_x" = 23, "under_y" = 15, "stock_x" = 22, "stock_y" = 12)
+
+	accuracy_mult_unwielded = 0.5 //Heavy and unwieldy; you don't one hand this.
+	scatter_unwielded = 100 //Heavy and unwieldy; you don't one hand this.
+	damage_falloff_mult = 0.25
+	fire_delay = 3
+	var/mode_index = 1
+	var/static/list/datum/lasrifle/base/mode_list = list(
+						/datum/lasrifle/base/standard,
+						/datum/lasrifle/base/auto,
+						/datum/lasrifle/base/overcharge,
+						/datum/lasrifle/base/heat,
+						/datum/lasrifle/base/spread)
```

Comment:
sort out the indent

Reply:
A static list makes you lose a lot on flexibility. Use a global list instead.

Reply:
they aren't being instanced and the list doesnt change, static is the right choice here

Reply:
The ways these datums are used are basically unmanageable global lists.
What's their advantage?
Both could be substituted by a referenced global list at no cost.

Reply:
Static gives you no advantages over a referenced global list and lowers flexibility.
It allows for no variants nor badminery.